### ISSUE TITLE
[PR#5160] minio runs in FS mode wrongly

### ISSUE THREAD

**Issue by balamurugana** [minio/minio#5522]
When running `minio server http://node{1...4}/home/bala/tmp/t/sda1` where node1, node2, node3 and node4 names are pointing to different IP of same machine makes minio to run in FS mode.
```
$ python -m json.tool sda1/.minio.sys/format.json 
{
    "format": "fs",
    "fs": {
        "version": "2"
    },
    "version": "1"
}
```

**Comment by harshavardhana** [comment]
There is a bug in current master as well 

```
minio server http://node{1..4}/tmp/t/sda1

panic: runtime error: index out of range

goroutine 1 [running]:
github.com/minio/minio/cmd.newObjectLayer(0x0, 0x0, 0x0, 0x2, 0xc4202bdcf0, 0x0, 0xc4202006c0)
	/home/harsha/mygo/src/github.com/minio/minio/cmd/server-main.go:263 +0x310
github.com/minio/minio/cmd.serverMain(0xc420229340)
	/home/harsha/mygo/src/github.com/minio/minio/cmd/server-main.go:222 +0x7b4
github.com/minio/minio/vendor/github.com/minio/cli.HandleAction(0xf344a0, 0x1158610, 0xc420229340, 0xc42000f300, 0x0)
	/home/harsha/mygo/src/github.com/minio/minio/vendor/github.com/minio/cli/app.go:499 +0x7c
github.com/minio/minio/vendor/github.com/minio/cli.Command.Run(0x110f9db, 0x6, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1129653, 0x1c, 0x0, ...)
	/home/harsha/mygo/src/github.com/minio/minio/vendor/github.com/minio/cli/command.go:214 +0xa99
github.com/minio/minio/vendor/github.com/minio/cli.(*App).Run(0xc4201bdba0, 0xc420080060, 0x6, 0x6, 0x0, 0x0)
	/home/harsha/mygo/src/github.com/minio/minio/vendor/github.com/minio/cli/app.go:260 +0x758
github.com/minio/minio/cmd.Main(0xc420080060, 0x6, 0x6)
	/home/harsha/mygo/src/github.com/minio/minio/cmd/main.go:149 +0x8d
main.main()
	/home/harsha/mygo/src/github.com/minio/minio/main.go:71 +0x108
```

**Comment by harshavardhana** [comment]
For some reason the endpoint.go returns empty Endpoints while the function indeed has success. 

```
minio server http://node{1..4}/tmp/t/sda1
:9000 [] mode-server-xl <nil>
```

**Comment by harshavardhana** [comment]
> When running minio server http://node{1...4}/home/bala/tmp/t/sda1 where node1, node2, node3 and node4 names are pointing to different IP of same machine makes minio to run in FS mode.

The reason this is happening @balamurugana the following code figures out that all endpoints are local and tries to a construct a list where all the paths are same pointing to same disk. In master it fails but in PR#5160 the endpoint path is picked as is which leads it to believe that its an FS endpoint. Will fix this in current master which would address this in PR 5160 as well.

```
   // All endpoints are pointing to local host                                                                                                                          
         if len(endpoints) == localEndpointCount {                                                                                                                            
                 // If all endpoints have same port number, then this is XL setup using URL style endpoints.                                                                  
                 if len(localPortSet) == 1 {                                                                                                                                  
                         if len(localServerAddrSet) > 1 {                                                                                                                     
                                 // TODO: Eventhough all endpoints are local, the local host is referred by different IP/name.                                                
                                 // eg '172.0.0.1', 'localhost' and 'mylocalhostname' point to same local host.                                                               
                                 //                                                                                                                                           
                                 // In this case, we bind to 0.0.0.0 ie to all interfaces.                                                                                    
                                 // The actual way to do is bind to only IPs in uniqueLocalHosts.                                                                             
                                 serverAddr = net.JoinHostPort("", serverAddrPort)                                                                                            
                         }                                                                                                                                                    
                                                                                                                                                                              
                         endpointPaths := endpointPathSet.ToSlice()                                                                                                           
                         endpoints, _ = NewEndpointList(endpointPaths...)                                                                                                   
                         setupType = XLSetupType                                                                                                                              
                         return serverAddr, endpoints, setupType, nil                                                                                                         
                 }                                              
```

**Comment by harshavardhana** [comment]
https://github.com/minio/minio/pull/5523 - this PR should fix the issue for PR #5160 as well.

**Comment by harshavardhana** [comment]
Fixed in #5160 PR after rebasing on #5523 

```
minio server http://node{1...4}/home/bala/tmp/t/sda1

Trace: 1: cmd/server-main.go:121:cmd.serverHandleCmdArgs()
       2: cmd/server-main.go:171:cmd.serverMain()
       3: vendor/github.com/minio/cli/app.go:499:cli.HandleAction()
       4: vendor/github.com/minio/cli/command.go:214:cli.Command.Run()
       5: vendor/github.com/minio/cli/app.go:260:cli.(*App).Run()
       6: cmd/main.go:149:cmd.Main()
       7: main.go:71:main.main()
[2018-02-15T09:28:40.017217356Z] [FATAL] Invalid command line arguments server=‘:9000’, args=[http://node{1...4}/home/bala/tmp/t/sda1] (path '/home/bala/tmp/t/sda1' cannot be served by different address on same server)
```
